File: smart_farm/src/api/alertEvent.ts
import { get, http } from '@/utils/request'
import type { PageInfo } from '@/api/user'

export interface AlertEvent {
  id: number
  userId: number
  username?: string
  nickname?: string
  plotId?: number
  plotName?: string
  plotCode?: string
  deviceId?: number
  deviceName?: string
  deviceCode?: string
  batchId?: number
  batchNo?: string
  alertType: number
  alertTitle: string
  alertContent?: string
  metricCode?: string
  metricName?: string
  metricValue?: number
  metricUnit?: string
  thresholdMin?: number
  thresholdMax?: number
  alertLevel?: number
  sourceType?: string
  sourceId?: number
  processStatus?: number
  handlerId?: number
  handleTime?: string
  handleResult?: string
  triggerTime?: string
  status?: number
  remark?: string
  createTime?: string
  updateTime?: string
}

export interface AlertEventListParams {
  plotName?: string
  alertType?: number
  alertLevel?: number
  processStatus?: number
  sourceType?: string
  startDate?: string
  endDate?: string
  pageNum: number
  pageSize: number
}

interface ApiResult<T> {
  code: number
  message: string
  data: T
}

export const listAlertEvents = (params: AlertEventListParams) => {
  return get<ApiResult<PageInfo<AlertEvent>>>('/alert-event/list', params as unknown as Record<string, unknown>)
}

export const getAlertEventById = (id: number) => {
  return get<ApiResult<AlertEvent>>(`/alert-event/${id}`)
}

export const updateAlertProcessStatus = (id: number, processStatus: number, handleResult?: string) => {
  return http<ApiResult<AlertEvent>>({
    url: `/alert-event/${id}/process-status`,
    method: 'put',
    params: { processStatus, handleResult },
  })
}

export const deleteAlertEvent = (id: number) => {
  return http<ApiResult<void>>({ url: `/alert-event/${id}`, method: 'delete' })
}

export const batchDeleteAlertEvents = (ids: number[]) => {
  return http<ApiResult<number>>({ url: '/alert-event/batch', method: 'delete', data: ids })
}

export const getAlertEventStatistics = () => {
  return get<ApiResult<Record<string, number>>>('/alert-event/statistics')
}

File: smart_plant/src/main/java/com/smart_plant/smart_plant/entity/AlertEvent.java
package com.smart_plant.smart_plant.entity;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class AlertEvent {

    private Long id;

    private Long userId;

    private String username;

    private String nickname;

    private Long plotId;

    private String plotName;

    private String plotCode;

    private Long deviceId;

    private String deviceName;

    private String deviceCode;

    private Long batchId;

    private String batchNo;

    private Integer alertType;

    private String alertTitle;

    private String alertContent;

    private String metricCode;

    private String metricName;

    private BigDecimal metricValue;

    private String metricUnit;

    private BigDecimal thresholdMin;

    private BigDecimal thresholdMax;

    private Integer alertLevel;

    private String sourceType;

    private Long sourceId;

    private String dedupKey;

    private Integer processStatus;

    private Long handlerId;

    private LocalDateTime handleTime;

    private String handleResult;

    private LocalDateTime triggerTime;

    private Integer status;

    private String remark;

    private LocalDateTime createTime;

    private LocalDateTime updateTime;
}
